aliyun/src/main/java/org/apache/iceberg/aliyun/AliyunClientFactories.java

```diff
@@ -81,17 +88,82 @@ private static AliyunClientFactory loadClientFactory(
   }
 
   static class DefaultAliyunClientFactory implements AliyunClientFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultAliyunClientFactory.class);
     private AliyunProperties aliyunProperties;
 
     DefaultAliyunClientFactory() {}
 
+    /**
+     * Check if RRSA environment variables are present. RRSA requires
+     * ALIBABA_CLOUD_OIDC_PROVIDER_ARN, ALIBABA_CLOUD_ROLE_ARN and ALIBABA_CLOUD_OIDC_TOKEN_FILE to
+     * be set.
+     */
+    boolean isRrsaEnvironmentAvailable() {
+      String oidcProviderArn = System.getenv("ALIBABA_CLOUD_OIDC_PROVIDER_ARN");
+      String roleArn = System.getenv("ALIBABA_CLOUD_ROLE_ARN");
+      String oidcTokenFile = System.getenv("ALIBABA_CLOUD_OIDC_TOKEN_FILE");
+      return oidcProviderArn != null
+          && !oidcProviderArn.isEmpty()
+          && roleArn != null
+          && !roleArn.isEmpty()
+          && oidcTokenFile != null
+          && !oidcTokenFile.isEmpty();
+    }
+
     @Override
     public OSS newOSSClient() {
       Preconditions.checkNotNull(
           aliyunProperties,
           "Cannot create aliyun oss client before initializing the AliyunClientFactory.");
 
-      if (Strings.isNullOrEmpty(aliyunProperties.securityToken())) {
+      String endpoint = aliyunProperties.ossEndpoint();
+
+      // Check if RRSA environment is available
+      if (isRrsaEnvironmentAvailable()) {
+        try {
+          LOG.info(
+              "Detected RRSA environment variables, creating OSS client with RRSA credentials for endpoint: {}",
+              endpoint);
+
+          // Use OIDCRoleArnCredentialProvider directly with built-in caching and auto-refresh
+          final OIDCRoleArnCredentialProvider oidcProvider =
+              OIDCRoleArnCredentialProvider.builder().build();
+
+          CredentialsProvider ossCredProvider =
+              new CredentialsProvider() {
+                private volatile Credentials currentCredentials;
+
+                @Override
+                public void setCredentials(Credentials credentials) {}
+
+                @Override
+                public Credentials getCredentials() {
+                  try {
+                    LOG.debug("Getting credentials using RRSA");
+                    // getCredentials() returns cached credentials and auto-refreshes when needed
+                    CredentialModel cred = oidcProvider.getCredentials();
+                    long expirationSeconds = 0;
+                    if (cred.getExpiration() > 0) {
+                      expirationSeconds =
+                          (cred.getExpiration() - System.currentTimeMillis()) / 1000;
```

Comment:
What's the time unit for the `cred.getExpiration` ?  Is it second or milliseconds ?

Reply:
It's milliseconds.  Inside OIDCRoleArnCredentialProvider.class,  We can find out it was set via `long expiration = ParameterHelper.getUTCDate(result.get("Expiration")).getTime();`, and getTime() have comment like "Returns the number of milliseconds since January 1, 1970, 00:00:00 GMT represented by this Date object."